pdns/lwres.cc

```diff
@@ -327,12 +327,21 @@ LWResult::Result asyncresolve(const ComboAddress& ip, const DNSName& domain, int
       Socket s(ip.sin4.sin_family, SOCK_STREAM);
 
       s.setNonBlocking();
-      ComboAddress local = pdns::getQueryLocalAddress(ip.sin4.sin_family, 0);
+      // v6 tcp does not seem to have fast open
+      if (ip.sin4.sin_family == AF_INET && SyncRes::s_tcp_fast_open_connect) {
```

Comment:
Based on https://www.spinics.net/lists/netdev/msg416206.html I believe `TCP_FASTOPEN_CONNECT` will work on IPv6 sockets as well.

Reply:
I'm not really sure. On my buster machines I see this:
```
# sudo sysctl -a | grep fast
net.ipv4.tcp_fastopen = 3
net.ipv4.tcp_fastopen_blackhole_timeout_sec = 3600
```
No mention of TFO for v6.

Reply:
I'm pretty sure these settings control both the IPv4 and IPv6 sides, and lives in `net.ipv4` for historical reasons, but I have not actually verified this.

Reply:
I will verify